src/libraries/System.Data.Common/tests/System/Data/DataTableTest.cs

```diff
@@ -376,23 +377,18 @@ public void SelectOperators()
 
         }
 
-        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsBinaryFormatterSupported), nameof(PlatformDetection.IsNotInvariantGlobalization))]
-        public void DataColumnTypeSerialization()
+        [Fact]
+        public void SerializationFormat_Binary_does_not_work_by_default_on_DataTable()
         {
             DataTable dt = new DataTable("MyTable");
-            DataColumn dc = new DataColumn("dc", typeof(int));
-            dt.Columns.Add(dc);
-            dt.RemotingFormat = SerializationFormat.Binary;
+            Assert.Throws<InvalidEnumArgumentException>(() => dt.RemotingFormat = SerializationFormat.Binary);
+        }
 
-            DataTable dtDeserialized;
-            using (MemoryStream ms = new MemoryStream())
-            {
-                BinaryFormatter bf = new BinaryFormatter();
-                bf.Serialize(ms, dt);
-                ms.Seek(0, SeekOrigin.Begin);
-                dtDeserialized = (DataTable)bf.Deserialize(ms);
-            }
-            Assert.Equal(dc.DataType, dtDeserialized.Columns[0].DataType);
+        [Fact]
+        public void SerializationFormat_Binary_does_not_work_by_default_on_DataSet()
+        {
+            DataSet ds = new DataSet();
+            Assert.Throws<InvalidEnumArgumentException>(() => ds.RemotingFormat = SerializationFormat.Binary);
```

Comment:
Rather than deleting the old version of the test, should we keep it as a separate one that uses RemoteInvoke to run it in a separate process that has the flag set?

Reply:
We could - in another library I've also made the flag property writable in DEBUG builds in order to check this without RemoteInvoke. The only thing is that here we plan to completely remove the code path in .NET 8.0, so too much effort into testing might not be justified.

But if you'd like me to test via RemoteInvoke, are we already doing this somewhere else, for the infrastructure/sample?

Reply:
Thanks @stephentoub, did this.